modules/ckanext-ytp_main/ckanext/ytp/harvesters/sixodp_harvester.py

```diff
@@ -62,21 +47,57 @@ def parse_datetime(datetime_string):
     return None
 
 
-def sixodp_to_opendata_preprocess(package_dict):
-    groups = []
+def group_map():
+    def _evaluate(predicate, values):
+        if isinstance(predicate, str):
+            return predicate in values
+        else:
+            return predicate(values)
 
-    for group in package_dict.get('groups', []):
-        mapped_groups = GROUP_MAP.get(group.get('name'))
+    def _and(*predicates):
+        def f(values):
+            return all(_evaluate(p, values) for p in predicates)
+        return f
 
-        if mapped_groups is not None:
-            for mapped_group in mapped_groups:
-                log.info("Mapping Group %s => %s", group, mapped_group)
-                groups.append({'name': mapped_group})
-        else:
```

Comment:
The old implementation returned the group if it was not included in the map, but the new implmentation only returns groups that are in the map. Is this intended? Will it result in issues?

Reply:
The returning lines are under this actually :) (76, 77)

Reply:
Intended. We know specifically what groups we want to map into what and how. The previous implementation was meant to only optionally map some categories into others while defaulting to passthrough.